src/main/java/net/minecraftforge/common/crafting/NBTIngredientFuzzy.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.crafting;
+
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import net.minecraft.advancements.critereon.NbtPredicate;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.network.FriendlyByteBuf;
+
+public class NBTIngredientFuzzy extends Ingredient
+{
+    private final ItemStack stack;
+    private final NbtPredicate predicate;
+    protected NBTIngredientFuzzy(ItemStack stack)
+    {
+        super(Stream.of(new ItemValue(stack)));
+        this.stack = stack;
+        this.predicate = stack.hasTag() ? NbtPredicate.ANY : new NbtPredicate(stack.getShareTag());
+    }
+
+    @Override
+    public boolean test(@Nullable ItemStack input)
+    {
+        if (input == null)
+            return false;
+        //Can't use areItemStacksEqualUsingNBTShareTag because it compares stack size as well
```

Comment:
This comment potentially should be removed given there are other reasons it can't be used for one it is using the predicate which is not checked in the method the comment references.